pkg/query-service/app/clickhouseReader/reader.go

```diff
@@ -2583,39 +2583,73 @@ func (r *ClickHouseReader) GetFilteredSpansAggregates(ctx context.Context, query
 }
 
-func (r *ClickHouseReader) SetTTL(ctx context.Context, ttlParams *model.TTLParams) (*model.SetTTLResponseItem, *model.ApiError) {
-
-	switch ttlParams.Type {
+func (r *ClickHouseReader) SetTTL(ctx context.Context,
+	params *model.TTLParams) (*model.SetTTLResponseItem, *model.ApiError) {
 
+	var req, tableName string
+	switch params.Type {
 	case constants.TraceTTL:
-		// error is skipped, handled earlier as bad request
-		tracesDuration, _ := time.ParseDuration(ttlParams.Duration)
-		second := int(tracesDuration.Seconds())
-		query := fmt.Sprintf("ALTER TABLE default.%v MODIFY TTL toDateTime(timestamp) + INTERVAL %v SECOND", signozTraceTableName, second)
-		_, err := r.db.Exec(query)
-
-		if err != nil {
-			zap.S().Error(fmt.Errorf("error while setting ttl. Err=%v", err))
-			return nil, &model.ApiError{model.ErrorExec, fmt.Errorf("error while setting ttl. Err=%v", err)}
+		tableName = signozTraceTableName
+		req = fmt.Sprintf(
+			"ALTER TABLE default.%v MODIFY TTL toDateTime(timestamp) + INTERVAL %v SECOND DELETE",
+			tableName, params.DelDuration)
+		if len(params.ColdStorageVolume) > 0 {
+			req += fmt.Sprintf(", toDateTime(timestamp) + INTERVAL %v SECOND TO VOLUME '%s'",
+				params.ToColdStorageDuration, params.ColdStorageVolume)
 		}
 
 	case constants.MetricsTTL:
-		// error is skipped, handled earlier as bad request
-		metricsDuration, _ := time.ParseDuration(ttlParams.Duration)
-		second := int(metricsDuration.Seconds())
-		query := fmt.Sprintf("ALTER TABLE %v.%v MODIFY TTL toDateTime(toUInt32(timestamp_ms / 1000), 'UTC') + INTERVAL %v SECOND", signozMetricDBName, signozSampleName, second)
-		_, err := r.db.Exec(query)
-
-		if err != nil {
-			zap.S().Error(fmt.Errorf("error while setting ttl. Err=%v", err))
-			return nil, &model.ApiError{model.ErrorExec, fmt.Errorf("error while setting ttl. Err=%v", err)}
+		tableName = signozMetricDBName + "." + signozSampleName
+		req = fmt.Sprintf(
+			"ALTER TABLE %v MODIFY TTL toDateTime(toUInt32(timestamp_ms / 1000), 'UTC') + "+
+				"INTERVAL %v SECOND DELETE", tableName, params.DelDuration)
+		if len(params.ColdStorageVolume) > 0 {
+			req += fmt.Sprintf(", toDateTime(toUInt32(timestamp_ms / 1000), 'UTC')"+
+				" + INTERVAL %v SECOND TO VOLUME '%s'",
+				params.ToColdStorageDuration, params.ColdStorageVolume)
 		}
 
 	default:
-		return nil, &model.ApiError{model.ErrorExec, fmt.Errorf("error while setting ttl. ttl type should be <metrics|traces>, got %v", ttlParams.Type)}
+		return nil, &model.ApiError{model.ErrorExec,
+			fmt.Errorf("error while setting ttl. ttl type should be <metrics|traces>, got %v",
+				params.Type)}
+	}
+
+	// Set the storage policy for the required table. If it is already set, then setting it again
+	// will not a problem.
+	if len(params.ColdStorageVolume) > 0 {
+		policyReq := fmt.Sprintf("ALTER TABLE %s MODIFY SETTING storage_policy='tiered'", tableName)
+
+		zap.S().Info("Executing Storage policy request: %s\n", policyReq)
```

Comment:
> zap.S().Info(...

it should be `zap.S().Infof(...`.
Also, for I think this log should be a debug level.